src/services/middlewareServices/AuthorizationMiddlewareService.ts

```diff
@@ -47,8 +47,10 @@ export default class AuthorizationMiddlewareService {
       userId
     )
     return collaboratorType === CollaboratorRoles.Admin
```

Comment:
Semantically ok with this, but this took me a while to digest, nit would be to refactor this and use type guards. 
Assuming there will be a case where `collaboratorType.Admin` && `collaboratorRole === CollaboratorRoles.Contributor`, which is why this complexity is needed. Please change this logic if this assumption is wrong.

Reply:
Hmm yeah I agree this should be refactored, because otherwise this would get very complex once we introduce more roles beyond Admin and Contributor. The assumption you have is correct.